src/labone/core/value.py

```diff
@@ -280,15 +285,54 @@ def _capnp_value_to_python_value(
     raise ValueError(msg)
 
 
+def _numpy_vector_to_capnp_vector(
+    np_vector: np.ndarray,
+    *,
+    path: LabOneNodePath,
+    reflection: ReflectionServer,
+) -> capnp.lib.capnp._DynamicStructBuilder:
+    """Convert a numpy vector to a capnp vector.
+
+    Args:
+        np_vector: The numpy vector to convert.
+        path: The path of the node the vector belongs to.
+        reflection: The reflection server used for the conversion.
+
+    Returns:
+        The converted capnp vector.
+
+    LabOneCoreError: If the numpy type has no corresponding
+        VectorElementType.
+    """
+    np_data = np_vector
+    np_vector_type = np_vector.dtype
+    if np.iscomplexobj(np_vector) and "waveforms" in path.lower():
```

Comment:
In my opinion, `"waveforms" in path.lower()` is error prone and we should find a more solid way

Reply:
I agree. But given the fact that we, in the following PI, move the shf vector parsing to the data server, this is the best change. We can guarantee that no node containing waveforms does not need this conversion. 

The alternative would be a much more invasive change, making the value conversion dependent on the session. And to prevent this we exactly wanting to move this logic to the data server